colossus/src/main/scala/colossus/controller/Controller.scala

```diff
@@ -297,8 +300,11 @@ extends ConnectionHandler {
           }
           currentTrigger = Some(trigger)
         }
-        case Failure(reason) => {
-          //todo: what to do here?
+        //TODO: are the following 2 cases really exceptions..seems like they are part of the normal control flow from a controller's POV
```

Comment:
I'm pretty sure a PipeTerminated exception needs to close the connection, since it means the controller is streaming data into something, and that thing is dead, so the controller has no way to know what to do with the incoming data.  That exception should only be thrown when the situation is basically "Everybody panic!!"  

This does though add evidence that the PipeClosedException should not really be an exception, and just a normal return value.